crates/router/src/connector/noon/transformers.rs

```diff
@@ -819,13 +840,18 @@ pub struct NoonRefundResponseTransactions {
 }
 
 #[derive(Default, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct NoonRefundSyncResponseResult {
     transactions: Vec<NoonRefundResponseTransactions>,
 }
 
 #[derive(Default, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct RefundSyncResponse {
     result: NoonRefundSyncResponseResult,
+    result_code: u32,
+    class_description: String,
+    message: String,
```

Comment:
Aren't these optional fields? Do we get this in case of refund success scenario also?

Reply:
Noon send these values in every scenario. If any fields are not applicable in any case, they return empty string or 0.